### ISSUE TITLE
yarn got error in cygwin Error: Cannot find module 'E:\cygdrive\d\tools\yarn\bin\yarn'

### ISSUE THREAD

**Issue by gensmusic** [yarnpkg/yarn#819]
**Do you want to request a *feature* or report a *bug*?**
bug
**What is the current behavior?**

**If the current behavior is a bug, please provide the steps to reproduce.**
install yarn
open cygwin
yarn --verison
 Error: Cannot find module 'E:\cygdrive\d\tools\yarn\bin\yarn'
**What is the expected behavior?**
got the right version of yarn
**Please mention your node.js, yarn and operating system version.**
node.js 4.6.7, win7, in cygwin

**Comment by Daniel15** [comment]
Which installation method did you use to install Yarn?

**Comment by tomzx** [comment]
In my case, I use the `.msi` provided on `https://yarnpkg.com/en/docs/install`.

**Comment by jbe456** [comment]
Replacing the `yarn` file content into the `bin` folder by:

```
#!/bin/sh
basedir=$(dirname "$(echo "$0" | sed -e 's,\\,/,g')")

case `uname` in
    *CYGWIN*) basedir=`cygpath -w "$basedir"`;;
esac

node "$basedir/yarn.js" "$@"
ret=$?

exit $ret
```

did the trick for me.

**Comment by gensmusic** [comment]
@Daniel15 I use  I use the .msi provided on https://yarnpkg.com/en/docs/install

**Comment by Daniel15** [comment]
Good catch... `yarn` should probably be a shell script that handles Cygwin.

**Comment by gensmusic** [comment]
@Daniel15 I just use `alias yarn=yarn.cmd` to work it around.

**Comment by abrcdf1023** [comment]
@gensmusic You save my day, cheer.
However, can you explain why it work?

**Comment by Daniel15** [comment]
@gensmusic Thanks for pointing me in the right direction! This pull request should fix it: #1180
